File: presentation/03_Structs/structs.m
// Topic: Structs
// Define Vector2 and Dot Product

#import <Foundation/Foundation.h>

typedef struct {
    int x;
    int y;
} Vector2;

int dot(Vector2 a, Vector2 b) {
    return (a.x * b.x) + (a.y * b.y);
}

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        Vector2 v1 = {10, 20};
        Vector2 v2 = {5, 5};

        int result = dot(v1, v2);
        NSLog(@"Dot Product: %d", result);
    }
    return 0;
}

File: presentation/03_Structs/structs.swift
// Topic: Structs
// Define Vector2 and Dot Product

struct Vector2 {
    var x: Int
    var y: Int
}

func dot(_ a: Vector2, _ b: Vector2) -> Int {
    return (a.x * b.x) + (a.y * b.y)
}

let v1 = Vector2(x: 10, y: 20)
let v2 = Vector2(x: 5, y: 5)

let result = dot(v1, v2)
print("Dot Product: \(result)")
